superset/cli/test_db.py

```diff
@@ -0,0 +1,417 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+from __future__ import annotations
+
+import sys
+from collections import defaultdict
+from datetime import datetime
+from typing import Any, Callable
+
+import click
+import yaml
+from rich.console import Console
+from sqlalchemy import (
+    Column,
+    create_engine,
+    DateTime,
+    ForeignKey,
+    insert,
+    Integer,
+    MetaData,
+    select,
+    String,
+    Table,
+)
+from sqlalchemy.engine import Engine
+from sqlalchemy.exc import NoSuchModuleError
+
+from superset.databases.utils import make_url_safe
+from superset.db_engine_specs import load_engine_specs
+from superset.db_engine_specs.base import BaseEngineSpec
+from superset.db_engine_specs.lib import (
+    ADVANCED_FEATURES,
+    BASIC_FEATURES,
+    DATABASE_DETAILS,
+    diagnose,
+    LIMIT_METHODS,
+    NICE_TO_HAVE_FEATURES,
+)
+
+metadata_obj = MetaData()
+
+user = Table(
+    "user",
```

Comment:
To limit the risk of collisions with pre-existing tables and make it easier to identify the table if it's left dangling, maybe we could add some prefix, like `tmp_superset_test_table_user`.

Reply:
Ah, food point, let me fix that!